src/api/server.spec.ts

```diff
@@ -29,4 +41,24 @@ describe('makeApiServer', () => {
 
     expect(serverInstance).toBe(mockFastify);
   });
+
+  describe('makeApiServerPlugin', () => {
+    const mockEnvVariables = configureMockEnvVars();
+
+    test('Existing env variable should register awala routes', async () => {
```

Comment:
I think this title is too vague. Suggestion:

```suggestion
    test('Awala routes should be registered if middleware is set', async () => {
```